src/irmin-pack/unix/gc.ml

```diff
@@ -157,24 +159,26 @@ module Worker = struct
 
       (* Step 1. Open the files *)
       [%log.debug "GC: opening files in RO mode"];
-      let* fm = Fm.open_ro config in
+      let fm = Fm.open_ro config |> Errs.raise_if_error in
       Errors.finalise (fun _outcome ->
```

Comment:
`Errors.finalise` except both arguments to not raise errors.

Could you change the implementation of `finalise` to make it call the `finiliser` if `f` raised?

Reply:
very good point, thanks